src/tensors/cpu/wasm_intgemm_interface.h

```diff
@@ -0,0 +1,193 @@
+ */
+void int8PrepareB(const float* input_B,
+                  float scale,
+                  int8_t zero_point,
+                  bool is_input_transposed,
+                  Index output_rows,
+                  Index output_cols,
+                  int8_t* output);
+
+/**
+ * Prepare B for the Matrix Multiply routine from an already quantized, transposed and a
+ * CPU-independent format of B.
+ *
+ * B is prepared in a CPU-dependent format. This function is useful while using the quantized models
+ * that are stored in a CPU-independent format on the disk.
+ *
+ * @param[in]   input_B         An array representing the input 2-D matrix.
+ *                              Size of the array = `output_rows` * `output_cols`.
+ *                              Shape of the matrix: (`output_cols`, `output_rows`)
+ * @param[in]   output_rows     No. of rows of output (prepared B) matrix.
+ *                              It should be a multiple of 64.
+ * @param[in]   output_cols     No. of columns of output (prepared B) matrix.
+ *                              It should be a multiple of 8.
+ * @param[out]  output          An array representing the prepared B matrix in row-major format.
+ *                              Size of the array = `output_rows` * `output_cols`.
+ *                              Shape of the matrix: (`output_rows`, `output_cols`)
+ */
+void int8PrepareBQuantizedTransposed(const int8_t* input_B,
+                                     Index output_rows,
+                                     Index output_cols,
+                                     int8_t* output);
+
+/**
+ * Prepare A for the Matrix Multiply routine.
+ *
+ * It performs quantization on floating values.
+ * Please note that this interface might have architecture specific implementations.
+ *
+ * @param[in]   input_A        An array representing the input 2-D matrix in row-major format.
+ *                             Size of the array = `output_rows` * `output_cols`.
+ *                             Shape of the matrix: (`output_rows`, `output_cols`)
+ * @param[in]   scale          The scaling factor (for quantization)
+ * @param[in]   zero_point     The zero point (for quantization)
+ * @param[in]   output_rows    No. of rows of output (prepared A) matrix.
+ *                             No restriction on its size.
+ * @param[in]   output_cols    No. of columns of output (prepared A) matrix.
+ *                             It should be a multiple of 64.
+ * @param[out]  output         An array representing the prepared A matrix in row-major format.
+ *                             Size of the array = `output_rows` * `output_cols`.
+ *                             Shape of the matrix: (`output_rows`, `output_cols`)
+ */
+void int8PrepareA(const float* input_A,
+                  float scale,
+                  int8_t zero_point,
+                  Index output_rows,
+                  Index output_cols,
```

Comment:
Why is this called output_cols while the other one is called cols_B?

Reply:
I will change it to be consistent everywhere.